### DISCUSSION TITLE
Styles not being applied on components imported from an external package

### DISCUSSION THREAD

**Discussion by fmazanerdwallet** [tailwindlabs/tailwindcss#6056] [Help] [answered]
Given the following code:

```
import { Button } from "react-component-library"

const Application: FunctionComponent = () => {
    return (
        <>
            <Button label="Click on me" />
            <div className="border-4 border-light-blue-500 border-opacity-100"></div>
       </>
    )
}
```

The styles set on the `Button` are reflected on the DOM node on the browser inspector but are not being applied.

On the other hand, the styles set on the `div` are both being reflected and applied correctly.

Would someone be so kind to help me with this issue, please?

**Comment by RobinMalfait** [comment, marked as answer]
Hey! Thank you for your question!
Much appreciated! 🙏

I'm assuming that you are using `JIT` mode here (otherwise everything was generated for you – unless you purged).
However, JIT mode works by scanning your source files for Tailwind classes. This means that inside your `purge` paths you have to include the paths to your components form your library. `./node_modules/react-component-library/**/*.js` for example.

This means that you have to do that for every project you are using that library in.
Alternatively, you can also create a `tailwind.config.js` file inside your package that already contains the correct paths. If you have that, than you can use the `presets` features and include that config file.

More info: https://tailwindcss.com/docs/configuration#presets

**Reply by jquense** [reply]
@RobinMalfait does that work? presets don't merge `purge` settings according to docs (and my experience). so if you want to add your local project files to the JIT config it would override any preset settings i believe

**Reply by RobinMalfait** [reply]
Ah right I forgot about the part that they are not getting merged. However, what you _can_ do is expose those paths and merge them in yourself manually. For example:

```js
let paths = require('react-component-library/paths.js')

module.export = {
  purge: ['./src/**/*.js', ...paths],
  // rest of the config
}
```

**Reply by vinayraghu1** [reply]
This is great and it works but want to note that this doesn't purge the CSS as you would expect. If the library contains 100s of components, the CSS will include all classes used in all components even if you are importing just one in your app.

**Comment by fmazanerdwallet** [comment]
@RobinMalfait, I'm so grateful! Including the path on the `purge` list definitely worked!

I'll check what @jquense mentioned later, but since that involves modifying the package and testing it I think I'm going to delay it just for now.

Thank you very much for the help!

**Comment by jquense** [comment]
I raised this as a good improvement here a bit ago: https://github.com/tailwindlabs/tailwindcss/discussions/5208

**Reply by hasanm95** [reply]
@jquense could you please, share your code with me? I am facing the same problem but can't solve it.

**Comment by federicofazzeri** [comment]
hello, is this still the best option we have to purge components imported from a library? thanks!

**Comment by jbexx** [comment]
If you're working with JIT its helpful to know that the "purge" option has been renamed to "content".
https://tailwindcss.com/docs/upgrade-guide#configure-content-sources

**Reply by 0tuedon** [reply]
This doesn't work with storybook.

**Comment by Jared-Dahlke** [comment]
@RobinMalfait how can we consume an NPM package that has tailwind in an app that does not have tailwind?

**Comment by beautyfree** [comment]
how it can be done today?

**Reply by 0tuedon** [reply]
By importing it directly from the package can't something like node_modules/style.css this is just an example

**Reply by 0tuedon** [reply]
But your app has to use tailwind

**Reply by beautyfree** [reply]
> By importing it directly from the package can't something like node_modules/style.css this is just an example

are you sure? if I just import a component there are no styles in my project for the classes used in this component

**Reply by 0tuedon** [reply]
So you have to import it in the tailwind config as a path

**Reply by beautyfree** [reply]
yep, it works, thank you!

**Reply by 0tuedon** [reply]
Check [here](https://github.com/tailwindlabs/tailwindcss/discussions/6056#discussioncomment-1627523)
